### PR TITLE
Sort By Set

### LINKED ISSUE #146: Sort by set implementation
Needs sorting overhaul wrapped in #123 

DEPENDENT ON: #123

### PR DESCRIPTION
**Pull request by JakeWags**
### Does this PR close any open issues?
Closes #146 

### Give a longer description of what this PR addresses and why it's needed
This PR adds the ability to sort the UpSet plot by a set, aka, bringing a set to the top. See #113 for example context menu for sets.

The implemented behavior is based on the original Upset behavior here: [http://vcg.github.io/upset/](http://vcg.github.io/upset/)

This PR also removes any reference to the SortBy type, as this is only a string alias.

### Provide pictures/videos of the behavior before and after these changes (optional)
BEFORE: 
upset-sortbyset-before

AFTER:
upset-sortbyset-after

### Have you added or updated relevant tests?
- [X] Yes
- [ ] No changes are needed

### Have you added or updated relevant documentation?
- [X] Yes
- [ ] No changes are needed

### Are there any additional TODOs before this PR is ready to go?

TODOs:
- none

### CODE REVIEW

**Comment by JakeWags** [comment]
> One point of discussion here is that I'd like to add tests for UpSet plot interactions such as aggregate, sort, etc; however, it doesn't seem like there is a good way to determine the order of the subsets on the plot, as their selectors are based on the position. Maybe adding an HTML id to each rendered row could be the solution to this.

Addressed in https://github.com/visdesignlab/upset2/pull/358/commits/ab1a972bca924cc60ddf9eca412d1375a727c14c

File: packages/upset/src/provenance/index.ts [hunk 1]
```diff
@@ -60,7 +60,8 @@ const sortByAction = registry.register(
   'sort-by',
   (state, { sort, sortByOrder }) => {
     state.sortBy = sort;
-    state.sortByOrder = sortByOrder;
+    // should only be 'None' if sortBy is a Set
+    state.sortByOrder = sortByOrder || 'None';
```
Comment (NateLanza) [inline packages/upset/src/provenance/index.ts:64]: I think using `null` instead of `'none'` here is slightly more type-safe and makes sense semantically.

Comment (NateLanza) [reply packages/upset/src/provenance/index.ts:64]: Or does `state.sortByOrder` need to be a string?

**Review by NateLanza — APPROVED** [review]
A number of minor things that you can address if you feel the need; merge at your discretion. No blocking issues.

File: packages/upset/src/provenance/index.ts (hunk 1 above)
Comment (JakeWags) [reply packages/upset/src/provenance/index.ts:64]: You're right that `state.sortByOrder` is a string. Using 'None' to me makes sense as there is no order for  sorting by set.

**Pull request merged**